crowbar_framework/app/models/service_object.rb

```diff
@@ -669,6 +669,10 @@ def select_nodes_for_role(all_nodes, role, preferred_intended_role = nil)
       valid_nodes = preferred_all_nodes unless preferred_all_nodes.empty?
     end
 
+    if role_constraints[role] && role_constraints[role].has_key?("count")
+      valid_nodes = valid_nodes.take(role_constraints[role]["count"])
```

Comment:
Because we just need to call it if the value is above 1? ```role_constraints[role].has_key?("count")``` is even for values like ```-1``` true because it just checks for the key and not the value like my solution.